spec/requests/internal/configs_spec.rb

```diff
@@ -15,137 +17,166 @@
-        end.to change(SiteConfig, :rate_limit_image_upload).from(9).to(3)
-      end
-
-      it "updates rate_limit_email_recipient" do
-        expect do
-          post "/internal/config", params: { site_config: { rate_limit_email_recipient: 3 } }
-        end.to change(SiteConfig, :rate_limit_email_recipient).from(5).to(3)
+        expect { post "/internal/config", params: { site_config: { favicon_url: expected_image_url }, confirmation: "Not proper" } }.to raise_error Pundit::NotAuthorizedError
       end
     end
 
-    describe "Google Analytics Reporting API v4" do
-      it "updates ga_view_id" do
-        post "/internal/config", params: { site_config: { ga_view_id: "abc" } }
-        expect(SiteConfig.ga_view_id).to eq("abc")
-      end
-
-      it "updates ga_fetch_rate" do
-        post "/internal/config", params: { site_config: { ga_fetch_rate: 3 } }
-        expect(SiteConfig.ga_fetch_rate).to eq(3)
-      end
-    end
-
-    describe "Mailchimp lists IDs" do
-      it "updates mailchimp_newsletter_id" do
-        post "/internal/config", params: { site_config: { mailchimp_newsletter_id: "abc" } }
-        expect(SiteConfig.mailchimp_newsletter_id).to eq("abc")
-      end
-
-      it "updates mailchimp_sustaining_members_id" do
-        post "/internal/config", params: { site_config: { mailchimp_sustaining_members_id: "abc" } }
-        expect(SiteConfig.mailchimp_sustaining_members_id).to eq("abc")
-      end
-
-      it "updates mailchimp_tag_moderators_id" do
-        post "/internal/config", params: { site_config: { mailchimp_tag_moderators_id: "abc" } }
-        expect(SiteConfig.mailchimp_tag_moderators_id).to eq("abc")
-      end
-
-      it "updates mailchimp_community_moderators_id" do
-        post "/internal/config", params: { site_config: { mailchimp_community_moderators_id: "abc" } }
-        expect(SiteConfig.mailchimp_community_moderators_id).to eq("abc")
-      end
-    end
-
-    describe "Email digest frequency" do
-      it "updates periodic_email_digest_max" do
-        post "/internal/config", params: { site_config: { periodic_email_digest_max: 1 } }
-        expect(SiteConfig.periodic_email_digest_max).to eq(1)
-      end
-
-      it "updates periodic_email_digest_min" do
-        post "/internal/config", params: { site_config: { periodic_email_digest_min: 3 } }
-        expect(SiteConfig.periodic_email_digest_min).to eq(3)
-      end
-    end
-
-    describe "Tags" do
-      it "removes space suggested_tags" do
-        post "/internal/config", params: { site_config: { suggested_tags: "hey, haha,hoho, bobo fofo" } }
-        expect(SiteConfig.suggested_tags).to eq(%w[hey haha hoho bobofofo])
-      end
-
-      it "downcases suggested_tags" do
-        post "/internal/config", params: { site_config: { suggested_tags: "hey, haha,hoHo, Bobo Fofo" } }
-        expect(SiteConfig.suggested_tags).to eq(%w[hey haha hoho bobofofo])
+    context "when admin has full permissions" do
```

Comment:
```suggestion
    context "when admin has single-resource and full permissions" do
```
This is slightly different than just full permissions, right?

Reply:
Good call. But I'm going to make a different change instead of this exact commit.